pkg/registry/core/service/ipallocator/allocator.go

```diff
@@ -202,7 +206,7 @@ func (r *Range) allocate(ip net.IP, dryRun bool) error {
 	ok, offset := r.contains(ip)
 	if !ok {
 		// update metrics
-		clusterIPAllocationErrors.WithLabelValues(label.String(), "static").Inc()
+		r.metrics.incrementAllocationErrors(label.String(), "static")
 		return &ErrNotInRange{ip, r.net.String()}
 	}
 	if dryRun {
```

Comment:
I've just realized that we are incrementing the metrics without checking `dryRun`, 

cc: @thockin

Reply:
this PR seems to fix this too

Reply:
Thanks. I made it do not increment when `dryRun` is true.

https://github.com/kubernetes/kubernetes/compare/f2f925a67f91f496e3875b8a8131f7eb243e352e..187af7781a4d6a308ea7f73dd22d435575873396#diff-3f7680212eb37166b66a0f052153bb765c0d4b141af958d5b92543dda628d35fR206